getting_started/step_by_step/your_first_game.rst

```diff
@@ -245,15 +249,10 @@ or ``false`` if it isn't.
  .. code-tab:: gdscript GDScript
 
     func _process(delta):
-        var velocity = Vector2()  # The player's movement vector.
-        if Input.is_action_pressed("ui_right"):
-            velocity.x += 1
-        if Input.is_action_pressed("ui_left"):
-            velocity.x -= 1
-        if Input.is_action_pressed("ui_down"):
-            velocity.y += 1
-        if Input.is_action_pressed("ui_up"):
-            velocity.y -= 1
+        var velocity = Vector2() # The player's movement vector.
+        velocity.x = Input.get_action_strength("move_right") - Input.get_action_strength("move_left")
+        velocity.y = Input.get_action_strength("move_down") - Input.get_action_strength("move_up")
+
         if velocity.length() > 0:
```

Comment:
I don't like this at all. The original code was explicit and clear. Each input is shown to map to a direction. This now gets into something called "action strength" which is not at all apparent what it means.

Reply:
This code is actually superior, since it properly supports joysticks, and it's also shorter. It seems clear to me, it says that the X velocity is equivalent to the right button minus the left button.

Reply:
I get it, but it's is not superior if the user is using default input actions (which they should be in this *beginner* demo).